Src/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -120,107 +124,40 @@ public virtual GoogleCredential CreateScoped(IEnumerable<string> scopes)
-
-        /// <summary>Initializes a HTTP client.</summary>
-        protected abstract void Initialize(ConfigurableHttpClient httpClient);
-
-        /// <summary>Gets the current access token.</summary>
-        protected abstract TokenResponse Token { get; }
-
-        /// <summary>Requests refreshing the access token.</summary>
-        protected abstract Task<bool> RequestAccessTokenAsync(CancellationToken taskCancellationToken);
+        internal ICredential UnderlyingCredential { get { return credential; } }
 
         #region Factory methods
 
-        /// <summary>Creates a <c>GoogleCredential</c> wrapping a <see cref="ComputeCredential"/>.</summary>
-        internal static GoogleCredential FromCredential(ComputeCredential credential)
-        {
-            return new ComputeGoogleCredential(credential);
-        }
-
         /// <summary>Creates a <c>GoogleCredential</c> wrapping a <see cref="ServiceAccountCredential"/>.</summary>
         internal static GoogleCredential FromCredential(ServiceAccountCredential credential)
         {
             return new ServiceAccountGoogleCredential(credential);
         }
 
-        /// <summary>Creates a <c>GoogleCredential</c> wrapping a <see cref="UserCredential"/>.</summary>
-        internal static GoogleCredential FromCredential(UserCredential credential)
-        {
-            return new UserGoogleCredential(credential);
-        }
-
         #endregion
 
-        // TODO(jtattermush): Look into adjusting the API of ServiceAccountCredential, ComputeCredential
-        // and UserCredential so that they implement ITokenAccess. Then the boilerplate below will go away.
-
-        /// <summary>Wraps <c>ComputeCredential</c> as <c>GoogleCredential</c>.</summary>
-        internal class ComputeGoogleCredential : GoogleCredential
-        {
-            private readonly ComputeCredential credential;
-
-            public ComputeGoogleCredential(ComputeCredential credential)
-            {
-                this.credential = credential;
-            }
-
-            #region GoogleCredential overrides
-
-            internal override object UnderlyingCredential
-            {
-                get { return credential; }
-            }
-
-            protected override void Initialize(ConfigurableHttpClient httpClient)
-            {
-                credential.Initialize(httpClient);
-            }
-
-            protected override TokenResponse Token
-            {
-                get { return credential.Token; }
-            }
-
-            protected override Task<bool> RequestAccessTokenAsync(CancellationToken taskCancellationToken)
-            {
-                return credential.RequestAccessTokenAsync(taskCancellationToken);
-            }
-
-            #endregion
-        }
-
         /// <summary>Wraps <c>ServiceAccountCredential</c> as <c>GoogleCredential</c>.</summary>
         internal class ServiceAccountGoogleCredential : GoogleCredential
```

Comment:
I spent some time experimenting with replacing this with passing delegates providing special handling for IsCreateScopedRequired and Create Scoped to the constructor and the result didn't look good. It resulted in more code being added and it was pretty unclear because I was basically trying to reinvent inheritance.
Leaving ServiceAccountGoogleCredential class, but added a comment clarify why we are subclassing.